pkg/apis/monitoring/v1/types.go

```diff
@@ -373,15 +375,21 @@ type PrometheusList struct {
 // +kubebuilder:validation:Pattern:="(^0|([0-9]*[.])?[0-9]+((K|M|G|T|E|P)i?)?B)$"
 type ByteSize string
 
+// Duration is a valid time unit
+// Supported units: y, w, d, h, m, s, ms Examples: `30s`, `1m`, `1h20m15s`
+// +kubebuilder:validation:Pattern:="(^0|(([0-9]+)y)?(([0-9]+)w)?(([0-9]+)d)?(([0-9]+)h)?(([0-9]+)m)?(([0-9]+)s)?(([0-9]+)ms)?)$"
+type Duration string
+
 // PrometheusSpec is a specification of the desired behavior of the Prometheus cluster. More info:
 // https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
 // +k8s:openapi-gen=true
 type PrometheusSpec struct {
 	CommonPrometheusFields `json:",inline"`
 	// Time duration Prometheus shall retain data for. Default is '24h' if
-	// retentionSize is not set, and must match the regular expression `[0-9]+(ms|s|m|h|d|w|y)`
+	// retentionSize is not set, and must match the regular expression `0|[0-9]+(ms|s|m|h|d|w|y)`
 	// (milliseconds seconds minutes hours days weeks years).
-	Retention string `json:"retention,omitempty"`
+	// +kubebuilder:default:="24h"
```

Comment:
ditto

Reply:
I have added a todo on statefulset.go file to remove default